src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHost.cs

```diff
@@ -161,11 +161,16 @@ internal async Task RunAsyncCore(CancellationToken cancellationToken, WebAssembl
 
                     try
                     {
+                        // Here, we add each root component but don't await the returned tasks so that the
+                        // components can be processed in parallel.
                         foreach (var rootComponent in rootComponents)
                         {
-                            await renderer.AddComponentAsync(rootComponent.ComponentType, rootComponent.Parameters, rootComponent.Selector);
+                            _ = renderer.AddComponentAsync(rootComponent.ComponentType, rootComponent.Parameters, rootComponent.Selector);
```

Comment:
An alternative to doing something like this could be introducing a new method in `Renderer`, maybe `BeginRenderRootComponent`. This would do what `RenderRootComponentAsync` does except await quiescence. Then we could avoid allocating `Task`s we're not even using. This line would then look something like

```suggestion
                            renderer.AddComponent(rootComponent.ComponentType, rootComponent.Parameters, rootComponent.Selector);
```

Reply:
I thought RenderRootComponentAsync did already wait for quiescence, but maybe am misremembering (could check on Monday). 

What happens if we collect all the tasks returned here and then Task.WhenAll on them, rather than making WaitForQuiescence public?

Reply:
@SteveSandersonMS Maybe I didn't word my comment correctly - yes, `RenderRootComponentAsync` does wait for quiescence. I like the `Task.WhenAll` idea, though.

Reply:
Actually, I'm thinking it still might be better (?) to make an alternative to `RenderRootComponentAsync` that doesn't wait for quiescence, then we wait for quiescence after we've started rendering each component (so, similar to how it currently works but we're not throwing away a bunch of tasks). If we did `Task.WhenAll`, each task being awaited is really the same thing, since `WaitForQuiescence` waits for all pending tasks to complete, and the pending tasks are shared across all components. Is my understanding correct? Do you think `Task.WhenAll` would still be preferable?

Reply:
If it were me I'd err on the side of not adding further public API to the renderer, as it doesn't seem necessary given the option of `Task.WhenAll`. I know most of the time all the tasks in the collection will actually be the same instance, but that doesn't seem like a drawback to me - it's just how it needs to be to cover all possible scenarios.